### PR TITLE
api: Make contains check for entity name

### PR DESCRIPTION
**Pull request by kezz**
This PR modifies `Component#contains(Component)` to check for the name of an entity in a `ShowEntity` hover event and adds some tests.

### CODE REVIEW

**Comment by kezz** [comment]
Rebased following merge of #364.

File: api/src/test/java/net/kyori/adventure/text/TextComponentTest.java [hunk 1]
```diff
@@ -142,18 +142,6 @@ void testContains() {
     assertTrue(component.contains(child));
   }
 
-  // https://github.com/KyoriPowered/adventure/issues/363
```
Comment (kashike) [inline api/src/test/java/net/kyori/adventure/text/TextComponentTest.java:145]: Don't remove this comment when you move this method to the new class.

**Review by kashike — CHANGES_REQUESTED** [review]

File: api/src/test/java/net/kyori/adventure/text/TextComponentTest.java (hunk 1 above)
Comment (kezz) [reply api/src/test/java/net/kyori/adventure/text/TextComponentTest.java:145]: Do you want me to stick it at the top of the class? The test is kinda subsumed into all three of the new tests.

Comment (kashike) [reply api/src/test/java/net/kyori/adventure/text/TextComponentTest.java:145]: sure

**Review by kashike — APPROVED** [review]

**Pull request merged**